src/Symfony/Component/Yaml/Parser.php

```diff
@@ -427,6 +427,10 @@ private function doParse($value, $flags)
                     $value = '';
 
                     foreach ($this->lines as $line) {
+                        // If the indentation is not consistent at offset 0, it is to be considered as a ParseError
+                        if (0 === $this->offset && !$deprecatedUsage && rtrim($line) !== trim($line)) {
```

Comment:
What about using `0 === $this->offset && !$deprecatedUsage && isset($line[0]) && ' ' === $line[0]` here instead to avoid calling `(r)trim()`?

Reply:
And is the check for `$deprecatedUsage` really needed? Which test would fail without it?

Reply:
@xabbuh thank you very much for your review and your feedback. Your version would work of course, I'll update. 
The $deprecated check is only needed in 3.4 because of this test....
https://github.com/symfony/symfony/blob/3.4/src/Symfony/Component/Yaml/Tests/ParserTest.php#L1787-L1800 that checks that the deprecation message is thrown.
In 4.0, the exception is expected (https://github.com/symfony/symfony/blob/4.0/src/Symfony/Component/Yaml/Tests/ParserTest.php#L1689-L1702)

I'm not familiar with how you idealy deals with this...